File: csharp/unittest-practiceTests/Clases/QuesoManchegoTests.cs


using unittestpractice.Clases;
using Microsoft.VisualStudio.TestTools.UnitTesting;


namespace unittest_practiceTests.Clases
{
    
    [TestClass()]
    public class QuesoManchegoTests
    {
        //Asserts

        readonly QuesoManchego _quesoManchego = new QuesoManchego();

        [TestMethod()]
        public void TestCurrentTemperature()
        {
            _quesoManchego.SetCurrentTemperature(21);
            Assert.AreEqual(21, _quesoManchego.GetCurrentTemperature());
        }

        [TestMethod()]
        public void TestFalseMelt()
        {
            _quesoManchego.Melt(false);
            Assert.IsFalse(_quesoManchego.IsMelted());
        }
        [TestMethod()]
        public void TestTrueMelt()
        {
            _quesoManchego.Melt(true);
            Assert.IsTrue(_quesoManchego.IsMelted());
        }
        [TestMethod()]
        public void TestMelting()
        {
            Assert.AreEqual(25, _quesoManchego.GetMeltingTemperature());
        }
    }
}

File: python/quesadilla/Test/QuesoManchego_tests.py
import unittest
from main.QuesoManchego import QuesoManchego


class queso_Manchego_test(unittest.TestCase):

    def setUp(self):
        self.quesoManchego = QuesoManchego()

    def test_current_temperature(self):
        self.quesoManchego.set_current_temperature(21)
        self.assertEquals(21, self.quesoManchego.get_current_temperature())

    def test_false_melt(self):
        self.quesoManchego.melt(False)
        self.assertFalse(self.quesoManchego.is_melted())

    def test_true_melt(self):
        self.quesoManchego.melt(True)
        self.assertTrue(self.quesoManchego.is_melted())

    def test_melting(self):
        self.assertEquals(20, self.quesoManchego.get_melting_temperature())
